src/main/java/book/store/repository/impl/BookRepositoryImpl.java

```diff
@@ -1,47 +1,51 @@
 package book.store.repository.impl;
 
+import book.store.exception.DataProcessingException;
 import book.store.model.Book;
 import book.store.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
 import java.util.List;
-import org.hibernate.Session;
-import org.hibernate.SessionFactory;
-import org.hibernate.Transaction;
-import org.springframework.beans.factory.annotation.Autowired;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Repository;
 
+@RequiredArgsConstructor
 @Repository
 public class BookRepositoryImpl implements BookRepository {
-    private final SessionFactory sessionFactory;
-
-    @Autowired
-    public BookRepositoryImpl(SessionFactory sessionFactory) {
-        this.sessionFactory = sessionFactory;
-    }
+    private static final String SAVING_BOOK_FAILED = "Can't save the book to DB, book: ";
+    private final EntityManagerFactory entityManagerFactory;
 
     @Override
     public Book save(Book book) {
-        Session session = null;
-        Transaction transaction = null;
-        try {
-            session = sessionFactory.openSession();
-            transaction = session.beginTransaction();
-            session.persist(book);
+        EntityTransaction transaction = null;
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            transaction = entityManager.getTransaction();
+            transaction.begin();
+            entityManager.persist(book);
             transaction.commit();
         } catch (Exception e) {
-            throw new RuntimeException("Can't store the book to DB", e);
-        } finally {
-            if (transaction != null) {
+            if (transaction != null && transaction.isActive()) {
                 transaction.rollback();
```

Comment:
do not use try with resources, cause entityManager will not be available after throwing an exception

Reply:
Also, why did you decide to use EntityManager, not Session?

Reply:
I decided to use `EntityManager` because it is part of the JPA standard, making the code more portable and less dependent on a specific ORM implementation like Hibernate. Should I use Sessions instead?

Reply:
Okay, thank you